app/util/score_calculator.py

```diff
@@ -0,0 +1,104 @@
+import typing
+from app.model.document import DocumentEdit, Mention, Relation
+from app.model.similarity_score import SimilarityScoreResponse
+from app.util.utils import all_edits_contain_same_tokens
+
+
+class ScoreCalculator:
+    def calc_score(self, document0: DocumentEdit, document1: DocumentEdit):
+        if not (
+            all_edits_contain_same_tokens(
+                [document0.document.tokens, document1.document.tokens]
+            )
+        ):
+            raise ValueError(
+                "Tokens in the different edits of the document are not the same."
+            )
+        mention_score = _calc_mention_score(
+            mention_list_0=document0.mentions, mention_list_1=document1.mentions
+        )
+        common_mentions_0, common_mentions_1 = _get_common_mention_indices(
+            mention_list_0=document0.mentions, mention_list_1=document1.mentions
+        )
+        relations_by_mentions_0 = _get_relations_by_mentions(
+            document0.relations, common_mentions_0
+        )
+        relations_by_mentions_1 = _get_relations_by_mentions(
+            document1.relations, common_mentions_1
+        )
+        considered_relation_quote = 0
+        if (len(document0.relations) + len(document1.relations)) != 0:
+            considered_relation_quote = (
+                len(relations_by_mentions_0) + len(relations_by_mentions_1)
+            ) / (len(document0.relations) + len(document1.relations))
+        relation_score = _calc_relation_score(
+            relation_list_0=relations_by_mentions_0,
+            relation_list_1=relations_by_mentions_1,
+        )
+        similarity_score_response = SimilarityScoreResponse(
+            mention_score=mention_score,
+            considered_relation_quote=considered_relation_quote,
+            relation_score=relation_score,
+        )
+        return similarity_score_response
+
+
+def _calc_mention_score(
+    mention_list_0: typing.List[Mention], mention_list_1: typing.List[Mention]
+):
+    true_positives = sum(
+        any(m0.equals(m1) for m1 in mention_list_1) for m0 in mention_list_0
+    )
+    return _calc_f1_score(
+        length_0=len(mention_list_0),
+        length_1=len(mention_list_1),
+        true_positives=true_positives,
+    )
+
+
+def _calc_relation_score(
+    relation_list_0: typing.List[Relation], relation_list_1: typing.List[Relation]
```

Comment:
Siehe Kommentar an der Datei hier als Beispiel:
```suggestion
    actual_relations: typing.List[Relation], predicted_relations: typing.List[Relation]
```